### PR TITLE
Console apps in Windows can raise OSError, effects system must be OS agnostic

### PR DESCRIPTION
**Pull request by Araq**
Reverts nim-lang/Nim#15874

### CODE REVIEW

**Comment by ringabout** [comment]
```nim
# test.nim
import terminal

writeStyled("hello", {styleBright})
```

Is this expected in windows(or `nim c -r --app:console test.nim`)?

```nim
D:\QQPCmgr\Desktop\Nim>nim c -r -d:consoleapp test.nim
Hint: used config file 'C:\Users\blue\.choosenim\toolchains\nim-1.4.0\config\nim.cfg' [Conf]
Hint: used config file 'C:\Users\blue\.choosenim\toolchains\nim-1.4.0\config\config.nims' [Conf]
.......................
C:\Users\blue\.choosenim\toolchains\nim-1.4.0\lib\pure\terminal.nim(40, 56) template/generic instantiation from here
C:\Users\blue\.choosenim\toolchains\nim-1.4.0\lib\pure\terminal.nim(903, 17) Error: can raise an unlisted exception: ref OSError
```

File: lib/pure/terminal.nim [hunk 1]
```diff
@@ -37,10 +37,7 @@ type
 
 var gTerm {.threadvar.}: owned(PTerminal)
 
-when defined(windows) and defined(consoleapp):
-  proc newTerminal(): owned(PTerminal) {.gcsafe, raises: [OSError].}
-else:
-  proc newTerminal(): owned(PTerminal) {.gcsafe, raises: [].}
+proc newTerminal(): owned(PTerminal) {.gcsafe, raises: [].}
```
Comment (Araq) [inline lib/pure/terminal.nim]:
```suggestion
proc newTerminal(): owned(PTerminal) {.gcsafe, raises: [OSError].}
```

**Comment by Araq** [comment]
The version that I just pushed is the correct one IMO.

**Comment by ringabout** [comment]
`OSError` is raised if and only if `consoleapp` is defined and in windows. 

If effects system must be OS and conditional variables agnostic, I agree on this change.
But could we change the title of this pull request? It should be called "effects system must be OS agnostic" etc.

**Comment by Araq** [comment]
> If effects system must be OS and conditional variables agnostic, I agree on this change.

Correct.

**Comment by ringabout** [comment]
It failed:
```
2020-11-09T12:06:11.9749061Z Failure: reNimcCrash
2020-11-09T12:06:11.9750596Z $ D:\a\1\s\bin\nim.exe c --hints:on -d:testing --nimblePath:tests/deps --nimCache:nimcache\tests\stdlib\tterminal_12759.nim_0d61f8370cad1d412f80b84d143e1257 --stdout --hint[Path]:off --hint[Processing]:off tests\stdlib\tterminal_12759.nim
2020-11-09T12:06:11.9752106Z Hint: used config file 'D:\a\1\s\config\nim.cfg' [Conf]
2020-11-09T12:06:11.9753063Z Hint: used config file 'D:\a\1\s\config\config.nims' [Conf]
2020-11-09T12:06:11.9753942Z Hint: used config file 'D:\a\1\s\tests\config.nims' [Conf]
2020-11-09T12:06:11.9755751Z tterminal_12759.nim(7, 22) template/generic instantiation from here
2020-11-09T12:06:11.9756925Z tterminal_12759.nim(8, 21) Error: can raise an unlisted exception: OSError
```
Tests should be changed to:
```nim
import terminal

proc test() {.raises:[IOError, OSError, ValueError].} =
  setBackgroundColor(stdout, bgRed)

test()
```

So this change may be a breaking change if users use effect system?

For example:

if users did something like this before, it will fail:
```nim
import terminal

proc test() {.raises:[].} =
  writeStyled("hello", {styleBright})

test()
```

```
D:\QQPCmgr\Desktop\Nim\test.nim(3, 22) template/generic instantiation from here
D:\QQPCmgr\Desktop\Nim\test.nim(4, 14) Error: can raise an unlisted exception: OSError
```

**Comment by Araq** [comment]
> So this change may be a breaking change if users use effect system?

Yeah. Probably slightly better not to change anything then.

**Pull request closed without merging**